deepchem/feat/smiles_tokenizer.py

```diff
@@ -0,0 +1,235 @@
+    """
+
+    def __init__(
+            self,
+            vocab_file: str='',
+            # unk_token="[UNK]",
+            # sep_token="[SEP]",
+            # pad_token="[PAD]",
+            # cls_token="[CLS]",
+            # mask_token="[MASK]",
+            **kwargs
+    ):
+
+        """Constructs a SmilesTokenizer.
+
+        Parameters
+        ----------
+        vocab_file: str
+            Path to a SMILES character per line vocabulary file.
+            Default vocab file is found in deepchem/feat/tests/data/vocab.txt
+        """
+
+        super().__init__(vocab_file, **kwargs)
+        # take into account special tokens in max length
+        self.max_len_single_sentence = self.max_len - 2
+        self.max_len_sentences_pair = self.max_len - 3
+
+        if not os.path.isfile(vocab_file):
+            raise ValueError(
+                "Can't find a vocab file at path '{}'.".format(vocab_file)
+            )
+        self.vocab = load_vocab(vocab_file)
+        self.highest_unused_index = max(
+            [
+                i for i, v in enumerate(self.vocab.keys())
+                if v.startswith("[unused")
+            ]
+        )
+        self.ids_to_tokens = collections.OrderedDict(
+            [(ids, tok) for tok, ids in self.vocab.items()]
+        )
+        self.basic_tokenizer = BasicSmilesTokenizer()
+        self.init_kwargs["max_len"] = self.max_len
+
+    @property
+    def vocab_size(self):
+        return len(self.vocab)
+
+    @property
+    def vocab_list(self):
+        return list(self.vocab.keys())
+
+    def _tokenize(self, text):
+        """
+        Tokenize a string into a list of tokens.
+
+        Parameters
+        ----------
+        text: str
+
+        """
+
+        split_tokens = [token for token in self.basic_tokenizer.tokenize(text)]
+        return split_tokens
+
+    def _convert_token_to_id(self, token):
+        """
+        Converts a token (str/unicode) in an id using the vocab.
+
+        Parameters
+        ----------
+        token: str
+        """
+        return self.vocab.get(token, self.vocab.get(self.unk_token))
+
+    def _convert_id_to_token(self, index):
+        """
+        Converts an index (integer) in a token (string/unicode) using the vocab.
+
+        Parameters
+        ----------
+        index: int
+
+        """
+        return self.ids_to_tokens.get(index, self.unk_token)
+
+    def convert_tokens_to_string(self, tokens):
+        """ Converts a sequence of tokens (string) in a single string.
+
+        Parameters
+        ----------
+        tokens: List[str]
+
+        """
+        out_string = " ".join(tokens).replace(" ##", "").strip()
+        return out_string
+
+    def add_special_tokens_ids_single_sequence(self, token_ids):
+        """
+        Adds special tokens to the a sequence for sequence classification tasks.
+        A BERT sequence has the following format: [CLS] X [SEP]
+
+
```

Comment:
Could you add "Parameters" sections to this method and the following methods?

Reply:
done!